### ISSUE TITLE
Training does not work

### ISSUE THREAD

**Issue by JackieQiang** [cjbayron/c-rnn-gan.pytorch#1]
Thank you very much for publishing the pytorch version of C-RNN-GAN. But when I was training, I found that training didn't work. G's loss is very high and doesn't drop, what parameters do I need to adjust to ensure normal training?
![Screen Shot 2019-10-12 at 12 01 59 PM](https://user-images.githubusercontent.com/10878226/66694697-065ffd80-ece9-11e9-816d-10b7d5c0eabd.png)

**Comment by cjbayron** [comment]
I am getting same behavior. This is why I mentioned in [Status](https://github.com/cjbayron/c-rnn-gan.pytorch/#status) that I have yet to train a generator that can output good results.

I created a sample [code](https://github.com/cjbayron/c-rnn-gan.pytorch/blob/master/train_simple.py) (with number sequences as input instead of MIDI) to investigate how to resolve this. What I've found out so far is that pre-training (both G and D) helps a lot. After 10 pretraining epochs and ~200 training epochs, this sample code can fairly mimic the input sequence.

I've applied the improvements in [train.py](https://github.com/cjbayron/c-rnn-gan.pytorch/blob/master/train.py) but it did not work the same. Even after pretraining, it seems Discriminator is "too good" for the Generator to fool.

I'm currently looking if this is an issue in hyperparameters, or if I have to apply other training techniques (e.g. freezing). I'll get back to you as soon as I get meaningful results.

**Comment by JackieQiang** [comment]
Appreciate you being replied.

**Comment by cjbayron** [comment]
Hi @JackieQiang , I have updated master branch. Training is a lot better now, but can still be unstable.

**Comment by JackieQiang** [comment]
Thank you very much, I will try it tomorrow. @cjbayron

**Comment by cjbayron** [comment]
@JackieQiang I will close this issue soon, as I have indicated in [Status](https://github.com/cjbayron/c-rnn-gan.pytorch/#status) my findings regarding this. Kindly inform if there further concerns :)
